File: Baekjoon/Class1/29.10818.py
import sys
input = sys.stdin.readline

min = 10000001
max = -10000001
_ = int(input())
for num in list(map(int, input().split())):
    if num < min: min = num
    if num > max: max = num
print(min, max)

'''
_ = input()
nums = list(map(int, input().split()))
print(min(nums), max(nums))
'''

File: Baekjoon/Class1/29.10818.cpp
#include <iostream>
using namespace std;

int main(void) {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	int n, input, min = 10000001, max = -10000001;
	cin >> n;
	while (n--) {
		cin >> input;
		if (input < min) min = input;
		if (input > max) max = input;
	}
	cout << min << ' ' << max;
	return 0;
}